osquery/filesystem/windows/fileops.cpp

```diff
@@ -501,47 +496,49 @@ static AclObject modifyAcl(PACL acl,
    * that deal with viewing or modifying the ACL (such as File Explorer).
    */
 
-  DWORD i = 0;
-  PACE_HEADER entry = nullptr;
+  unsigned long i = 0;
+  LPVOID void_ent = nullptr;
   for (i = 0; i < info.AceCount; i++) {
-    if (!::GetAce(acl, i, (LPVOID*)&entry)) {
+    if (!GetAce(acl, i, &void_ent)) {
       return std::move(AclObject());
     }
 
+    auto entry = static_cast<PACE_HEADER>(void_ent);
     if ((entry->AceFlags & INHERITED_ACE) == INHERITED_ACE) {
       break;
     }
 
+    auto allowed_ace = reinterpret_cast<ACCESS_ALLOWED_ACE*>(entry);
```

Comment:
Gunna move these down closer to where they're actually used to avoid the `cast`ing if possible.